ddtrace/contrib/kafka/patch.py

```diff
@@ -149,7 +162,15 @@ def traced_produce(func, instance, args, kwargs):
         span.set_tag_str(COMPONENT, config.kafka.integration_name)
         span.set_tag_str(SPAN_KIND, SpanKind.PRODUCER)
         span.set_tag_str(kafkax.TOPIC, topic)
-        span.set_tag_str(kafkax.MESSAGE_KEY, ensure_text(message_key, errors="replace"))
+
+        if isinstance(instance, _SerializingProducer):
+            if hasattr(instance, "_key_serializer"):
+                serialized_key = serialize_key(instance, topic, message_key, headers)
+                if serialized_key is not None:
+                    span.set_tag_str(kafkax.MESSAGE_KEY, serialized_key)
+        else:
+            span.set_tag_str(kafkax.MESSAGE_KEY, message_key)
```

Comment:
Do we need the `ensure_text(....)` function here?

Reply:
`set_tag_str ` already uses `ensure_text` internally within the function